### PR TITLE
Implement source IP in REST authenticator

### PR DESCRIPTION
**Pull request by robklg**
Solves https://github.com/bolcom/libunftp/issues/491

### CODE REVIEW

File: crates/unftp-auth-rest/src/lib.rs [hunk 1]
```diff
@@ -44,23 +46,105 @@ pub struct Builder {
 }
 
 impl Builder {
+    /// Creates a new `Builder` instance with default settings.
     ///
+    /// This method initializes a new builder that you can use to configure and
```
Comment (hannesdejager) [inline crates/unftp-auth-rest/src/lib.rs:51]: Thanks for the comments

**Review by hannesdejager — APPROVED** [review]
LGTM

**Review by fbs — APPROVED** [review]

File: crates/unftp-auth-rest/src/lib.rs [hunk 2]
```diff
@@ -107,37 +192,46 @@ impl Builder {
...
-        let username_json = encode_string_json(username);
-        let password_json = encode_string_json(password);
-        let body = self.fill_encoded_placeholders(&self.body, &username_json, &password_json);
+        let url = self.fill_encoded_placeholders(&self.url, &username_url, &password_url, &source_ip_url);
 
-        // FIXME: need to clone too much, just to keep tokio::spawn() happy, with its 'static requirement. is there a way maybe to work this around with proper lifetime specifiers? Or is it better to just clone the whole object?
-        let method = self.method.clone();
-        let selector = self.selector.clone();
-        let regex = self.regex.clone();
+        let username_json = serde_json::to_string(username).map_err(|e| AuthenticationError::ImplPropagated(e.to_string(), None))?;
```
[the remark this answers is not in the document]
Comment (robklg) [reply crates/unftp-auth-rest/src/lib.rs:224]: Definitely!

File: crates/unftp-auth-rest/src/lib.rs [hunk 3]
```diff
@@ -107,37 +192,46 @@ impl Builder {
...
+        let trimmed_username_json = username_json.trim_matches('"');
```
[the remark this answers is not in the document]
Comment (robklg) [reply crates/unftp-auth-rest/src/lib.rs:225]: Totally

**Pull request merged**